packages/story-editor/src/app/story/effects/useLoadStory.js

```diff
@@ -82,7 +83,11 @@ function loadStory(storyId, post, restore, clearHistory, globalConfig) {
   // If there are no pages, create empty page.
   const storyData =
     storyDataRaw && migrate(storyDataRaw, storyDataRaw.version || 0);
-  const pages = storyData?.pages?.length > 0 ? storyData.pages : [createPage()];
+  let pages = storyData?.pages?.length > 0 ? storyData.pages : [createPage()];
+
+  if (storyData?.fonts && Object.keys(storyData?.fonts).length >= 1) {
+    pages = populateElementFontData(pages, storyData?.fonts);
```

Comment:
Ah, okay, so when loading the fonts, it will populate the data and when saving, it will remove it just for the saving part? In this case do we even need the migration at all for this? Since it will be cleaned when saving the story anyway? A few lines above the migration removes the font data and here it's added back again.

Q: Is there a plan to follow up with not having to populate the text elements and just using the information from the page directly? Wondering why the reloading-cleaning process is necessary if we have the information on the page directly (Probably you had a discussion about it already somewhere).

Reply:
Right -- added some notes in the PR description but that's easy to miss.

Save 

-- remove font props from each element just keep font family 
-- write the font full font font into to story fonts


Reload 
- put the data back

Here's the JSON that gets saved if that's helpful

https://github.com/GoogleForCreators/web-stories-wp/pull/12625#issuecomment-1320961376


>  Is there a plan to follow up with not having to populate the text elements and just using the information from the page directly? Wondering why the reloading-cleaning process is necessary if we have the information on the page directly (Probably you had a discussion about it already somewhere).

I ran into a bunch of issues around 
- timing for font data to be available (i.e. console log to calcFontMeterics in one render sometimes it would have the data and other times it wouldn't)
- how to access the font data in some cases --- when no story provider exists

I was also getting concerned with the amount of code changes and the possibility of breaking existing stories.

--------------------------------------------------------------------------------

I think this PR moves us along on the storage end i.e. database storage + easier to read template files without the redundant font data.  

It also keeps the passing of data / function signatures the same so we should be safe for backwards compatibility etc...

In my mind with this PR it's more about storage less about how we push the data around.

Reply:
Alright. In this case, what's the reason for adding the migration at all? 
Currently what's happening when loading the story is:
- [migration runs and removes the font data](https://github.com/GoogleForCreators/web-stories-wp/pull/12625/files/baf785d35f5830972d422c689b399bb2ec374815#diff-ea54c3a9b6ee8086ade8e30de7af7c6204697eac9c91d7c5173271c967c53975R85)
- [font data is added back](https://github.com/GoogleForCreators/web-stories-wp/pull/12625/files/baf785d35f5830972d422c689b399bb2ec374815#diff-ea54c3a9b6ee8086ade8e30de7af7c6204697eac9c91d7c5173271c967c53975R89)

Looks like the data is added back right after the migration runs, doesn't seem there's any point doing that? Thoughts?

Reply:
Yes - - I agree I think it can just be removed - that said was unsure if for existing data that hasn't been saved might need it to get the story fonts level data available?  I haven't really dug into how the migrations work / run.

Reply:
Going to remove it.

Reply:
Migration just runs when the story is loaded meaning it doesn't influence the DB until the user actually saves the story.

Reply:
Maybe leave a comment about it to the code -- to explain what's happening when adding back / removing font data? Since it doesn't influence functionality then it might not be clear why it's being done?

Reply:
Added some comments